File: scaffold/src/main/java/top/xuqingquan/web/publics/IndicatorController.java
package top.xuqingquan.web.publics;

import top.xuqingquan.web.nokernel.BaseIndicatorSpec;

public interface IndicatorController {

    void progress(android.webkit.WebView v, int newProgress);

    void progress(com.tencent.smtt.sdk.WebView v, int newProgress);

    BaseIndicatorSpec offerIndicator();

    void showIndicator();

    void setProgress(int newProgress);

    void finish();
}

File: scaffold/src/main/java/top/xuqingquan/web/publics/IndicatorController.kt
package top.xuqingquan.web.publics

import top.xuqingquan.web.nokernel.BaseIndicatorSpec

interface IndicatorController {

    fun progress(v: android.webkit.WebView, newProgress: Int)

    fun progress(v: com.tencent.smtt.sdk.WebView, newProgress: Int)

    fun offerIndicator(): BaseIndicatorSpec?

    fun showIndicator()

    fun setProgress(newProgress: Int)

    fun finish()
}
